File: Leetcode Challenge/June/Max Consecutive Ones III.cpp
class Solution {
public:
    int longestOnes(vector<int>& nums, int k) {
        
        int i = 0, j = 0, n = nums.size(), countZeros = 0, ans = INT_MIN;
        
        while(j < n){
            
            if(nums[j] == 0){
                countZeros++;
            }
            while(countZeros > k){
                
                if(nums[i] == 0){
                    
                    countZeros--;      
                }
                
                i++;
            }
            
            ans=max(ans, j-i + 1);
            
            j++;
            
        }
        
        return ans;
        
    }
};

File: Leetcode Challenge/June/Python/Max Consecutive Ones III.py
class Solution:
    def longestOnes(self, A: List[int], K: int) -> int:
        
        start=0
        end=0
        countzeros=0
        maxv=0
        while end<len(A):
            if A[end]==0:
                countzeros+=1
            
            while countzeros>K:
                if A[start]==0:
                    countzeros-=1

                start+=1
               
            
            maxv=max(maxv,end-start+1)
            end+=1
            
        
        return maxv
